### DISCUSSION TITLE
Status of the project

### DISCUSSION THREAD

**Discussion by amsuu** [amsuu/INDP#64] [Announcements]
# Status of the project
### INDP dropped !?!1!?!?!11!? INDP is here!!?!!1!!111?!1?!1??
No. The project still remains, as it has been since its start, under development. By me exclusively (not by choice). This has lead to some terrible looking code, a nearly unfollowable codebase and delayed development due to mismanagement of priorities. I am currently focusing on the learn pages in order to make this a suitable learning material to cite in discussions on the discord, etc. After that will follow likely an implementation of Az.ts from @bt2901 in order to implement some sort of automatic declension to ease the quiz creation process. After that will follow maybe some quiz creation framework/interface for developers to make new quizzes (this may also invite contributors). After which I will implement the (currently dead) Level-3 ("word in context" -- which includes prepositions and more missing words like adjectives and the prepositions themselves -- though maybe WITH prepos-s and WITHOUT prepos-s should be seperate levels) and Level-4 ("comprehension" -- a sentence like "Ja znam russkujų učitelkų bulgarskogo" and then a reader has to answer "what's her nationality?" and "what language is she teaching?" -- thank you to @bt2901 again for the suggestion!)

#### In SASS/SCSS stylesheets:
  - - [x] implement mixins
- - [x] fix up logic and style hierarchies
- - [ ] remove unnecessary variables and definitions
- - [ ] set up better defaults and remove redundant styles
- - [ ] modularize the styles better, creating a better hierarchy, in which the parent is not required to stylize the child (in some cases where this is a bad thing)
- - [ ]  generally document my decisions better
#### In TS/TypeScript logic:
- - [ ] take out functions and types which have more general uses outside from specific components and into services and interfaces.
- - [x] clean up some logic and comment better
- - [ ] make more seamless and less verbose integrations with the HTML
- - [x]  generally document my decisions better
#### In HTML/Angular structures:
- - [x] better modularization of components and more (and less) abstraction (where needed!).
- - [ ] cleaner HTML with less verbosity (no matter how smart the implementations are and/or make you look, @amsuu)
- - [ ] better class names and more `id`s instead of `class`es where appropriate

## In Conclusion
I hope the publicity of this post will help me stay on track. And I hope that these changes will make for a more readable and more accessible codebase. After all,

#### _- me_

**Comment by amsuu** [comment]
#65 will deal with the BIGGEST issues which need urgent fixing, developer-experience-wise. After that I may invite more contributors if they wish to assist in the project.

**Comment by amsuu** [comment]
![RDT_20241022_1559436575764112453532711.jpg](https://github.com/user-attachments/assets/dfe1d029-66ef-433a-a1df-0fb2768fa688)

**Comment by amsuu** [comment]
#73 #74 as well
